src/twisted/conch/test/test_ckeygen.py

```diff
@@ -63,6 +65,46 @@ def setUp(self):
         self.patch(sys, 'stdout', self.stdout)
 
 
+
+    def test_runecdsa(self):
+        filename = self.mktemp()
+        subprocess.call(['ckeygen', '-t', 'ecdsa', '-f', filename, '--no-passphrase'])
+        privKey = Key.fromFile(filename)
+        pubKey = Key.fromFile(filename + '.pub')
+        self.assertEqual(privKey.type(), 'EC')
+        self.assertTrue(pubKey.isPublic())
+
+
+
+    def test_rundsa(self):
+        filename = self.mktemp()
+        subprocess.call(['ckeygen', '-t', 'dsa', '-f', filename, '--no-passphrase'])
+        privKey = Key.fromFile(filename)
+        pubKey = Key.fromFile(filename + '.pub')
+        self.assertEqual(privKey.type(), 'DSA')
+        self.assertTrue(pubKey.isPublic())
+
+
+
+    def test_runrsa(self):
+        filename = self.mktemp()
+        subprocess.call(['ckeygen', '-t', 'rsa', '-f', filename, '--no-passphrase'])
+        privKey = Key.fromFile(filename)
+        pubKey = Key.fromFile(filename + '.pub')
+        self.assertEqual(privKey.type(), 'RSA')
+        self.assertTrue(pubKey.isPublic())
+
+
+    # def test_runBadKeytype(self):
+    #     filename = self.mktemp()
+    #     with self.assertRaises(SystemExit) as em:
+    #         subprocess.call(['ckeygen', '-t', 'foo', '-f', filename])
+    #     self.assertEqual(
+    #         'Key type was foo, must be one of: rsa, dsa, ecdsa',
+    #         em.exception.args[0])
```

Comment:
@Lukasa travis build is green but I still don't have full coverage because of this commented out test which I cannot figure out. Can you please take a look?